src/test/java/com/microsoft/sqlserver/testframework/DBConnection.java

```diff
@@ -28,71 +28,110 @@
-			setInternal(connection);
-		} catch (SQLException ex) {
-			fail(ex.getMessage());
-		} catch (ClassNotFoundException ex) {
-			fail(ex.getMessage());
-		}
-	}
-
-	@Override
-	void setInternal(Object internal) {
-		this.internal = internal;
-	}
-
-	/**
-	 * 
-	 * @return Statement wrapper
-	 */
-	public DBStatement createStatement() {
-		try {
-			DBStatement dbstatement = new DBStatement(this);
-			return dbstatement.createStatement();
-		} catch (SQLException ex) {
-			fail(ex.getMessage());
-		}
-		return null;
-	}
+    // TODO: add Isolation Level
+    // TODO: add auto commit
+    // TODO: add connection Savepoint and rollback
+    // TODO: add additional connection properties
+    // TODO: add DataSource support
+    private SQLServerConnection connection = null;
 
+    /**
+     * establishes connection using the input
+     * 
+     * @param connectionString
+     */
+    public DBConnection(String connectionString) {
+        super(null, null, "connection");
+        getConnection(connectionString);
+    }
 
     /**
-     * clsoe connection
+     * establish connection
+     * 
+     * @param connectionString
+     */
+    void getConnection(String connectionString) {
+        try {
+            connection = PrepUtil.getConnection(connectionString);
+            setInternal(connection);
+        }
+        catch (SQLException ex) {
+            fail(ex.getMessage());
+        }
+        catch (ClassNotFoundException ex) {
+            fail(ex.getMessage());
+        }
+    }
+
+    @Override
+    void setInternal(Object internal) {
+        this.internal = internal;
+    }
+
+    /**
+     * 
+     * @return Statement wrapper
+     */
+    public DBStatement createStatement() {
+        try {
+            DBStatement dbstatement = new DBStatement(this);
+            return dbstatement.createStatement();
+        }
+        catch (SQLException ex) {
+            fail(ex.getMessage());
+        }
+        return null;
+    }
+
+    /**
+     * 
+     * @param type
+     * @param concurrency
+     * @return
+     * @throws SQLException
+     */
+    public DBStatement createStatement(int type, int concurrency) throws SQLException {
+        DBStatement dbstatement = new DBStatement(this);
+        return dbstatement.createStatement(type, concurrency);
+
+    }
+
+    /**
+     * 
+     * @param rsType
+     * @return
+     * @throws SQLServerException
+     */
+    public DBStatement createStatement(DBResultSetTypes rsType) throws SQLServerException {
+        DBStatement dbstatement = new DBStatement(this);
+        return dbstatement.createStatement(rsType.resultsetCursor, rsType.resultSetConcurrency);
+    }
+
+    /**
+     * 
+     * @param query
+     * @return
+     * @throws SQLException
+     */
+    public DBPreparedStatement prepareStatement(String query) throws SQLException {
+        DBPreparedStatement dbpstmt = new DBPreparedStatement(this, internal, "preparedStatement");
+        return dbpstmt.prepareStatement(query);
```

Comment:
the variable internal passed here must be preparedStatement object. Can we have DBPreparedStatement(DBConnection dbConnection) constructor in DBPreparedStatement like the one in DBStatement? it will reduce the confusion